### PR TITLE
ENH add support for alpha dropout and SELU

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds support for alpha dropout and SELU activations.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,5 +21,5 @@
     install_requires=['numpy',
                       'scipy',
                       'scikit-learn~=0.19',
-                      'tensorflow~=1.3']
+                      'tensorflow~=1.6']
```
Comment (mheilman) [inline setup.py]: is < 1.6 still ok?

Why 1.6 rather than 1.7?

File: muffnn/mlp/base.py [hunk 2]
```diff
@@ -284,7 +284,11 @@ def _set_up_graph(self):
                            scope='layer_%d' % i, sparse_input=True)
             else:
                 if self.keep_prob != 1.0:
-                    t = tf.nn.dropout(t, keep_prob=self._keep_prob)
+                    if self.activation is tf.nn.selu:
+                        t = tf.contrib.nn.alpha_dropout(
```
Comment (mheilman) [inline muffnn/mlp/base.py]: it seems like it'd be better to have a `dropout_type` kwarg control this, unless it's the case that that alpha dropout could never be used except with selu activations and vice versa.

Comment (beckermr) [reply muffnn/mlp/base.py]: So alpha dropout is only meant for selu activations. The technique was invented in the paper on SELU and is designed explicitly for that. Thus I think this hard coding is better.

Comment (mheilman) [reply muffnn/mlp/base.py]: ah ok.  could you add a note to one of the docstrings?  i'm not sure whether the class docstring or method kwarg documentation is more appropriate.  probably the kwarg (e.g., "Setting this to `tf.nn.selu` will also cause alpha dropout to be used, implementing a Self-Normalizing Neural Network (Klambauer et al., 2017).").

**Comment by beckermr** [comment]
OK @mheilman - back to you!
